### PR TITLE
Move functions to helper and environment handle

### PR DESCRIPTION
**Pull request by srdjadzogaz**
Added header authorization validation 
Fixed post put and patch methods

### CODE REVIEW

File: helper.js [hunk 1]
```diff
@@ -1,57 +1,110 @@
...
   } catch (error) {
-    console.error('Error reading users_basic_auth.json:', error.message);
+    console.error("Error reading users_basic_auth.json:", error.message);
     return [];
   }
 }
 
 // Function to write user data to the JSON file
-function writeUsersToFile(usersFilePath) {
+function writeUsersToFile(usersFilePath, environment) {
```
Comment (vesnaatwork) [inline helper.js]: either remove env or userFilePath. the environment parameter isn't used

Comment (srdjadzogaz) [reply helper.js]: I have removed the env. It’s up thx ✋

File: index.js [hunk 2]
```diff
@@ -1,62 +1,70 @@
...
 
 // POST endpoint to add a new user
-app.post('/users', (req, res) => {
+app.post("/users", (req, res) => {
   const { name, age } = req.body;
+  const users = readUsersFromFile(usersFilePath);
+  const result = checkBasicAuthFromRequest(req);
+
+  if (!result) {
+    return res.status(400).send("Unauthorized access");
```
Comment (vesnaatwork) [inline index.js]: return 401

**Review by vesnaatwork — APPROVED** [review]
🚀

**Pull request merged**